src/python/foglamp/configuration_manager.py

```diff
@@ -0,0 +1,330 @@
+    query_template = """
+        SELECT 
+            configuration.value::json->'{}' as value
+        FROM 
+            foglamp.configuration
+        WHERE 
+            configuration.key='{}'
+    """
+    query_full = query_template.format(category_item, category_key)
+    async with aiopg.sa.create_engine(_connection_string) as engine:
+        async with engine.acquire() as conn:
+            async for row in conn.execute(text(query_full).columns(_configuration_tbl.c.value)):
+                return row.value
+
+async def _read_category_item_value(category_key, category_item):
+    query_template = """
+        SELECT 
+            configuration.value::json->'{}'->'Value' as value
+        FROM 
+            foglamp.configuration
+        WHERE 
+            configuration.key='{}'
+    """
+    query_full = query_template.format(category_item, category_key)
+    async with aiopg.sa.create_engine(_connection_string) as engine:
+        async with engine.acquire() as conn:
+            async for row in conn.execute(text(query_full).columns(_configuration_tbl.c.value)):
+                return row.value
+
+async def _update_item_value(category_key, category_item, item_value_replacement):
+    query_template = """
+            UPDATE foglamp.configuration 
+            SET value = jsonb_set(value, '{{{},Value}}', '"{}"') 
+            WHERE key='{}'
+        """
+    query_full = query_template.format(category_item, item_value_replacement, category_key)
+    async with aiopg.sa.create_engine(_connection_string) as engine:
+        async with engine.acquire() as conn:
+            await conn.execute(query_full)
+
+async def get_all_category_names():
+    """Get all category names in the FogLAMP system
+
+    Keyword Arguments:
+    None
+
+    Return Values:
+    a list of strings, each representing a unique category name
+    None
+
+    Side Effects:
+    None
+
+    Exceptions Raised:
+    Unknown
+
+    Restrictions and Usage:
+    A FogLAMP component calls this method to obtain all category names in the FogLamp system
+    """
+    try:
+        return await _read_all_category_keys()
+    except:
+        logging.getLogger(_logger_name).exception(
+            'Unable to read all category names')
+        raise
+
+
+async def get_all_category_items(category_name):
+    """Get a specified category's entire configuration.
+
+    Keyword Arguments:
+    category_name -- name of the category (required)
+
+    Return Values:
+    a JSONB dictionary with all items of a category's configuration
+    None
+
+    Side Effects:
+    None
+
+    Exceptions Raised:
+    Unknown
+
+    Restrictions and Usage:
+    A FogLAMP component calls this method to obtain a category's configuration
+    """
+    try:
+        return await _read_category_value(category_name)
+    except:
+        logging.getLogger(_logger_name).exception(
+            'Unable to get all category names based on category_name {}'.format(category_name))
```

Comment:
let interpolation overhead be handled by logging code itself? 

`("Unable to get all category names based on category_name : %s", category_name)`

NOTE: this comment is for logger string format only (vs .format which is our best practice)